### PR TITLE
Clarify contributing guide: PRs don't need everyone's re-approval before merging

### PR DESCRIPTION
**Pull request by nilsvu**
## Proposed changes

This is a proposal to reduce the number of times that reviewers have to go back and re-approve a PR. For example, when a PR has been approved but then needs a rebase or a clang-tidy fix I suggest it doesn't have to be re-approved by all reviewers if the core dev responsible for merging it determines the changes were sufficiently trivial. Of course the core dev can decide to ping the reviewers and ask for their explicit re-approval nonetheless, if they think that would be useful in the specific case.

When editing the review guidelines I found that they don't actually say that all reviewers have to approve again, so I just added a small note to clarify.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `major new feature` if appropriate.

### CODE REVIEW

**Review by wthrowe — CHANGES_REQUESTED** [review]
I'm fine with rebases and squashes and similar, but reviewers should still re-approve after comments have been addressed (at least if they resulted in non-trivial changes).

**Comment by nilsvu** [comment]
@wthrowe agreed. Do you think anything needs to be changed in the guidelines? Seems to me the line `Once all reviewers have approved the pull request, someone should ping (...)` covers it.

**Comment by wthrowe** [comment]
The new addition contradicts that: "The SpECTRE core developer will merge the pull request once all comments have been addressed...  The pull request does not need to be approved again by all reviewers."

**Comment by nilsvu** [comment]
To me the guidelines read: 1. All reviewers have to approve. 2. The core dev does a cursory review and merges without all reviewers having to re-approve.
Are you saying you want reviewers to re-approve when the cursory review leads to changes?

**Comment by wthrowe** [comment]
From the same document, any changes from the cursory review are supposed to address "major problems", so, yes, I think a reapproval should be required.

**Comment by nilsdeppe** [comment]
I would be okay with:
- all reviewers have approved all full reviewers (including the cursory review by the core-dev). I think it's important for reviewers to look at what the core-dev viewed as necessary additional changes (I can say that I always learn a lot from seeing what e.g. @wthrowe had for comments on PRs). The core-dev might also just have a bad idea and the other reviewers could catch that :)
- any minor changes after all reviewers (including core-dev) have approved, e.g. rebase to resolve conflict, fix clang-tidy complaint, missing include doesn't need everyone to re-approve. I think anything beyond a couple of lines should have unanimous re-approval, unless someone previously stated "I don't need to re-approve for any changes" (or something like that)

**Comment by wthrowe** [comment]
I approve of @nilsdeppe's proposal.

**Comment by nilsvu** [comment]
Sound good, reworded the guidelines. Please take another look.

**Review by wthrowe — APPROVED** [review]

**Review by nilsdeppe — APPROVED** [review]

**Review by kidder — APPROVED** [review]

**Pull request merged**
